### ISSUE TITLE
[BUG] 3D image at wrong location when `xflip = true`

### ISSUE THREAD

**Issue by bowenszhu** [JuliaPlots/Plots.jl#4304]
## Details

```julia
using Plots
plot([1, 2, 3], [2, 3, 1], [3, 1, 2], xflip = true)
```
yields
![p](https://user-images.githubusercontent.com/45696147/183863461-0e34314e-12c9-4d59-a18c-9c935e4d7a00.png)

### Backends

This bug occurs on ( insert `x` below )

Backend      | yes | no  | untested
-------------|-----|-----|---------
gr (default) |x  |     |
pyplot       |     |     |x
plotlyjs     |     |     |x
pgfplotsx    |     |     |x
unicodeplots |     |     |x
inspectdr    |     |     |x
gaston       |     |     |x

### Versions

Plots.jl version: v1.31.6
Backend  version (`]st -m <backend(s)>`):
Output of `versioninfo()`:
```
Julia Version 1.7.3
Commit 742b9abb4d (2022-05-06 12:58 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-12.0.1 (ORCJIT, skylake)
Environment:
  JULIA_EDITOR = code
  JULIA_NUM_THREADS = 6
```

**Comment by BeastyBlacksmith** [comment]
Might be related to #4232 

CC: @jheinen

**Comment by t-bltg** [comment]
This used to be correct:

**1.29.1**
https://docs.juliaplots.org/v1.29/gallery/gr/generated/gr-ref55/#gr_demo_55 ✔

**1.30.2**
https://docs.juliaplots.org/v1.30/gallery/gr/generated/gr-ref55/#gr_demo_55 ✘

**Comment by jheinen** [comment]
Is fixed in 0.66.1 - the BB build is waiting to be merged
